ducktape/services/service.py

```diff
@@ -158,7 +158,7 @@ def allocate_nodes(self):
                     "Service: %s, node.account: %s" % (self.__class__.__name__, str(node.account)))
             node.account.logger = self.logger
 
-        self.logger.debug("Successfully allocated %d nodes to %s" % (self.num_nodes, self.who_am_i()))
+        self.logger.debug("Successfully allocated %d nodes to %s" % (self.num_nodes, self.__class__.__name__))
```

Comment:
Using `who_am_i` before `self` is registered with the service registry triggered the following circular recursion:

```
Traceback (most recent call last):
  File "/Users/geoffreyanderson/Google_Drive/Confluent_code/cp-system-test/ducktape/ducktape/tests/runner_client.py", line 121, in run
    self.setup_test()
  File "/Users/geoffreyanderson/Google_Drive/Confluent_code/cp-system-test/ducktape/ducktape/tests/runner_client.py", line 167, in setup_test
    self.test.setup()
  File "/Users/geoffreyanderson/Google_Drive/Confluent_code/cp-system-test/ducktape/systests/cluster/test_remote_account.py", line 87, in setup
    self.service = GenericService(self.test_context, 1)
  File "/Users/geoffreyanderson/Google_Drive/Confluent_code/cp-system-test/ducktape/systests/cluster/test_remote_account.py", line 72, in __init__
    super(GenericService, self).__init__(context, num_nodes)
  File "/Users/geoffreyanderson/Google_Drive/Confluent_code/cp-system-test/ducktape/ducktape/services/service.py", line 64, in __init__
    self.allocate_nodes()
  File "/Users/geoffreyanderson/Google_Drive/Confluent_code/cp-system-test/ducktape/ducktape/services/service.py", line 166, in allocate_nodes
    self.logger.debug("Successfully allocated %d nodes to %s" % (self.num_nodes, self.who_am_i()))
  File "/Users/geoffreyanderson/Google_Drive/Confluent_code/cp-system-test/ducktape/ducktape/services/service.py", line 136, in who_am_i
    return self.service_id
  File "/Users/geoffreyanderson/Google_Drive/Confluent_code/cp-system-test/ducktape/ducktape/services/service.py", line 90, in service_id
    return "%s-%d-%d" % (self.__class__.__name__, self._order, id(self))
  File "/Users/geoffreyanderson/Google_Drive/Confluent_code/cp-system-test/ducktape/ducktape/services/service.py", line 112, in _order
    index = same_services.index(self)
  File "/Users/geoffreyanderson/Google_Drive/Confluent_code/cp-system-test/ducktape/ducktape/services/service.py", line 77, in __repr__
    return "<%s: %s>" % (self.who_am_i(), "num_nodes: %d, nodes: %s" %
  File "/Users/geoffreyanderson/Google_Drive/Confluent_code/cp-system-test/ducktape/ducktape/services/service.py", line 136, in who_am_i
    return self.service_id
  File "/Users/geoffreyanderson/Google_Drive/Confluent_code/cp-system-test/ducktape/ducktape/services/service.py", line 90, in service_id
    return "%s-%d-%d" % (self.__class__.__name__, self._order, id(self))
  File "/Users/geoffreyanderson/Google_Drive/Confluent_code/cp-system-test/ducktape/ducktape/services/service.py", line 112, in _order
    index = same_services.index(self)
  File "/Users/geoffreyanderson/Google_Drive/Confluent_code/cp-system-test/ducktape/ducktape/services/service.py", line 77, in __repr__
    return "<%s: %s>" % (self.who_am_i(), "num_nodes: %d, nodes: %s" %
  File "/Users/geoffreyanderson/Google_Drive/Confluent_code/cp-system-test/ducktape/ducktape/services/service.py", line 136, in who_am_i
    return self.service_id
  File "/Users/geoffreyanderson/Google_Drive/Confluent_code/cp-system-test/ducktape/ducktape/services/service.py", line 90, in service_id
    return "%s-%d-%d" % (self.__class__.__name__, self._order, id(self))
RuntimeError: maximum recursion depth exceeded while calling a Python object
```